src/RageSoundPosMap.cpp

```diff
@@ -133,38 +119,35 @@ std::int64_t pos_map_queue::Search( std::int64_t iSourceFrame, bool *bApproximat
 		return 0;
 	}
 
-	/* iSourceFrame is probably in pos_map.  Search to figure out what position
-	 * it maps to. */
+	// iSourceFrame is probably in pos_map.  Search to figure out what position it maps to.
 	std::int64_t iClosestPosition = 0, iClosestPositionDist = INT_MAX;
-	const pos_map_t *pClosestBlock = &*m_pImpl->m_Queue.begin(); /* print only */
+	
 	for (pos_map_t const &pm : m_pImpl->m_Queue)
 	{
+		// Loop over the queue until we know generally where iSourceFrame is
 		if( iSourceFrame >= pm.m_iSourceFrame &&
 			iSourceFrame < pm.m_iSourceFrame+pm.m_iFrames )
 		{
-			/* iSourceFrame lies in this block; it's an exact match.  Figure
-			 * out the exact position. */
-			int iDiff = int(iSourceFrame - pm.m_iSourceFrame);
-			iDiff = std::lrint( iDiff * pm.m_fSourceToDestRatio );
+			// If we are in the correct block, calculate its current position
+			std::int64_t iDiff = int64_t(iSourceFrame - pm.m_iSourceFrame);
+			iDiff = static_cast<int64_t>(( iDiff * pm.m_fSourceToDestRatio + 0.5 )); 
```

Comment:
Redundant brackets on this one - Or could be `static_cast<int64_t>((iDiff * ratio) + 0.5);`?